src/main/java/net/neoforged/neoforge/registries/DeferredRegister.java

```diff
@@ -331,60 +276,338 @@ public ResourceKey<? extends Registry<T>> getRegistryKey() {
+         * @param block The supplier for the block to create a {@link BlockItem} for.
+         * @return A {@link DeferredItem} that will track updates from the registry for this item.
+         * @see #registerBlockItem(String, Supplier, Item.Properties)
+         * @see #registerBlockItem(String, Holder, Item.Properties)
+         * @see #registerBlockItem(String, Holder)
+         * @see #registerBlockItem(DeferredHolder, Item.Properties)
+         * @see #registerBlockItem(DeferredHolder)
+         */
+        public DeferredItem<BlockItem> registerBlockItem(String name, Supplier<? extends Block> block) {
+            return this.registerBlockItem(name, block, new Item.Properties());
+        }
+
+        /**
+         * Adds a new {@link BlockItem} for the given {@link Block} to the list of entries to be registered and
+         * returns a {@link DeferredItem} that will be populated with the created item automatically.
+         *
+         * @param name       The new item's name. It will automatically have the {@linkplain #getNamespace() namespace} prefixed.
+         * @param block      The holder for the block to create a {@link BlockItem} for.
+         * @param properties The properties for the created {@link BlockItem}.
+         * @return A {@link DeferredItem} that will track updates from the registry for this item.
+         * @see #registerBlockItem(String, Supplier, Item.Properties)
+         * @see #registerBlockItem(String, Supplier)
+         * @see #registerBlockItem(String, Holder)
+         * @see #registerBlockItem(DeferredHolder, Item.Properties)
+         * @see #registerBlockItem(DeferredHolder)
+         */
+        public DeferredItem<BlockItem> registerBlockItem(String name, Holder<Block> block, Item.Properties properties) {
+            return this.register(name, key -> new BlockItem(block.value(), properties));
+        }
+
+        /**
+         * Adds a new {@link BlockItem} for the given {@link Block} to the list of entries to be registered and
+         * returns a {@link DeferredItem} that will be populated with the created item automatically.
+         * This method uses the default {@link Item.Properties}.
+         *
+         * @param name  The new item's name. It will automatically have the {@linkplain #getNamespace() namespace} prefixed.
+         * @param block The holder for the block to create a {@link BlockItem} for.
+         * @return A {@link DeferredItem} that will track updates from the registry for this item.
+         * @see #registerBlockItem(String, Supplier, Item.Properties)
+         * @see #registerBlockItem(String, Supplier)
+         * @see #registerBlockItem(String, Holder, Item.Properties)
+         * @see #registerBlockItem(DeferredHolder, Item.Properties)
+         * @see #registerBlockItem(DeferredHolder)
+         */
+        public DeferredItem<BlockItem> registerBlockItem(String name, Holder<Block> block) {
```

Comment:
This method (and other similar methods) cannot be invoked with a `DeferredHolder` because it implements both `Holder` and `Supplier`. We might want to make `Holder` implement `Supplier`, but it would then mean that `DeferredHolder<R, T>` cannot be converted to a `Supplier<T>` directly. (One would have to add `::get`).

Reply:
This should be resolved now with the two new overloads below

Reply:
There is no `(String, DeferredHolder)` overload.

Reply:
Fixed by removing the `(String, Holder, ...)` overloads.